### ISSUE TITLE
UI blocked on port updates

### ISSUE THREAD

**Issue by kieferrm** [microsoft/vscode-remote-release#4759]
Opening and closing notebooks leaves many ports in the ports view (separate issue). But the more ports are in the ports view the longer the ports view becomes unusable when there are changes. In the recording below, I close a notebook which changes the status of 12 ports. It takes 180s to get the ports view back to a workable state.

https://user-images.githubusercontent.com/4674940/112404959-92dc6100-8cce-11eb-9709-7379b386fccf.mov

**Comment by alexr00** [comment]
Looks like I'm just causing too many rerenders of the table since it is rendered for every port change.
